src/lib.rs

```diff
@@ -1052,3 +1052,37 @@ fn with_waker<F: FnOnce(&Waker) -> R, R>(f: F) -> impl Future<Output = R> {
         Poll::Ready(f(cx.waker()))
     })
 }
+
+fn _ensure_send_and_sync() {
+    use futures_lite::future::pending;
+
+    fn is_send<T: Send>(_: T) {}
+    fn is_sync<T: Sync>(_: T) {}
+
+    is_send::<Executor<'_>>(Executor::new());
+    is_sync::<Executor<'_>>(Executor::new());
+
+    let ex = Executor::new();
+    is_send(ex.run(pending::<()>()));
+    is_sync(ex.run(pending::<()>()));
+    is_send(ex.tick());
+    is_sync(ex.tick());
+
+    /// ```compile_fail
+    /// use async_executor::LocalExecutor;
+    /// use futures_lite::future::pending;
+    ///
+    /// fn is_send<T: Send>(_: T) {}
+    /// fn is_sync<T: Sync>(_: T) {}
+    ///
+    /// is_send::<LocalExecutor<'_>>(LocalExecutor::new());
+    /// is_sync::<LocalExecutor<'_>>(LocalExecutor::new());
+    ///
+    /// let ex = LocalExecutor::new();
+    /// is_send(ex.run(pending::<()>()));
+    /// is_sync(ex.run(pending::<()>()));
+    /// is_send(ex.tick());
+    /// is_sync(ex.tick());
+    /// ```
+    fn _negative_test() {}
```

Comment:
(I think we expect all of the is_*s here to fail, but actually even if just one of them fails, this will be marked as a compaile_fail.)